create-directories.bash

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+
+# Given a source directory, /source, and a target directory,
+# /target/foo/bar/bazz, we want to "clone" the target structure
+# from source into the target. Essentially, we want both
+# /source/target/foo/bar/bazz and /target/foo/bar/bazz to exist
+# on the filesystem. More concretely, we'd like to map
+# /state/etc/ssh/example.key to /etc/ssh/example.key
+#
+# To achieve this, we split the target's path into parts -- target, foo,
+# bar, bazz -- and iterate over them while accumulating the path
+# (/target/, /target/foo/, /target/foo/bar, and so on); then, for each of
+# these increasingly qualified paths we:
+#
+#   1. Ensure both /source/qualifiedPath and qualifiedPath exist
+#   2. Copy the ownership of the source path into the target path
+#   3. Copy the mode of the source path into the target path
+
+# Get inputs from command line arguments
```

Comment:
```suggestion
# Get inputs from command line arguments
if [ "$#" -neq 2 ]; then
    printf "\e[1;31mError: 'create-directories.bash' requires *two* args.\e[0m\n"
    exit 1
fi
```

Reply:
Added this, but with a `[[` test instead, which is easier to read and harder to mess up. I changed all other tests from `[` to `[[` too, to standardize. Also, instead of making the text red, I print it to `stderr`.

Reply:
Works for me :)